packages/kamino-sdk/src/Kamino.ts

```diff
@@ -3266,18 +3562,52 @@ export class Kamino {
     if (rebalanceKind.kind == RebalanceType.Manual.kind) {
       lowerPrice = rebalanceParams[0];
       upperPrice = rebalanceParams[1];
-    } else if (
-      rebalanceKind.kind == RebalanceType.PricePercentage.kind ||
-      rebalanceKind.kind == RebalanceType.PricePercentageWithReset.kind
-    ) {
-      let positionPrices = await this.getPriceRangePercentageBasedFromPool(
+    } else if (rebalanceKind.kind == RebalanceType.PricePercentage.kind) {
+      let { lowerPrice: posLowerPrice, upperPrice: posUpperPrice } = getPositionRangeFromPercentageRebalanceParams(
+        price,
+        rebalanceParams[0],
+        rebalanceParams[1]
+      );
+      lowerPrice = posLowerPrice;
```

Comment:
why here and below you override `lowerPrice` and `upperPrice`? couldn't you just take the result from the function?